launcher/icons/IconList.cpp

```diff
@@ -337,10 +337,16 @@ void IconList::installIcon(const QString& file, const QString& name)
     if (!fileinfo.isReadable() || !fileinfo.isFile())
         return;
 
-    if (!IconUtils::isIconSuffix(fileinfo.suffix()))
+    const QString sourceSuffix = fileinfo.suffix();
+    if (!IconUtils::isIconSuffix(sourceSuffix))
```

Comment:
To fix this, we can just go down the same path if either is bad
```suggestion
    const auto targetInfo = name.isEmpty() ? fileinfo : QFileInfo(name);
    if (!IconUtils::isIconSuffix(fileinfo.suffix()) || !Iconutils::isIconSuffix(targetInfo.suffix()))
```